### PR TITLE
JetBrains: improve settings UI

### PR DESCRIPTION
**Pull request by olafurpg**
- Clearer titles for sections: Authentication, Cody AI, Code Search
- Reorder groups to put Cody AI above Code Search
- Disabling Cody checkbox disables all checkboxes in the group
- Disabling Cody checkbox no longer sets autocomplete value to false. When you disable/enable Cody, the Cody-related setting values are the unchanged. Previously, all Cody values would be set to false.
- Automatically start/stop Agent based on Cody enabled setting

## Test plan

With `./gradlew -PenableAgent=true :runIde`:

- Cody + autocomplete is enabled, confirm autocomplete works.
- Disable Cody, confirm autocomplete checkbox is disabled but checked.
- Confirm autocomplete does not work.
- Enable Cody, disable autocomplete. Confirm autocomplete is still not working.
- Enable Cody, enable autocomplete, confirm autocomplete still works.

### CODE REVIEW

**Comment by olafurpg** [comment]
New settings UI titles 
CleanShot 2023-08-15 at 16 58 33@2x

Also, autocomplete+debug settings are checked even if they are disabled

CleanShot 2023-08-15 at 16 58 58@2x

File: client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/CodyAutoCompleteManager.java [hunk 1]
```diff
@@ -97,7 +97,7 @@ && isProjectAvailable(editor.getProject())
    */
   public void triggerAutoComplete(@NotNull Editor editor, int offset) {
     // Check if auto-complete is enabled via the config
-    if (!ConfigUtil.isCodyAutoCompleteEnabled()) {
+    if (!ConfigUtil.isCodyEnabled() || !ConfigUtil.isCodyAutoCompleteEnabled()) {
```
Comment (vdavid) [inline client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/CodyAutoCompleteManager.java:100]: I think we also need the same `ConfigUtil.isCodyEnabled()` check in line 86.

Comment (olafurpg) [reply client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/CodyAutoCompleteManager.java]: We shutdown the agent when cody is disabled so autocomplete isn't going to run anyways regardless if the `isAutocompleteEnabled()` code path is true.

**Review by vdavid — APPROVED** [review]
Much cleaner, thanks for the facelift + other improvements!

**Pull request merged**